src/components/Collection/Files/FilesSummaryGroup.tsx

```diff
@@ -0,0 +1,154 @@
+      sizeMap[idx].count += item.Count;
+    });
+
+    return map(sizeMap, (size, name) => (
+      `${size.count} ${name} (${prettyBytes(size.size, { binary: true })})`
+    )).join(' | ');
+  }, [group]);
+
+  const groupDetails = useMemo(() => (group.GroupName ? `${group.GroupName} (${group.GroupNameShort})` : '-'), [
+    group,
+  ]);
+  const videoDetails = useMemo(() => {
+    const conditions: string[] = [];
+    if (group.FileSource) {
+      conditions.push(group.FileSource.replace('BluRay', 'Blu-Ray'));
+    }
+    if (group.FileVersion) {
+      conditions.push(`v${group.FileVersion}`);
+    }
+    if (group.VideoBitDepth) {
+      conditions.push(`${group.VideoBitDepth}-bit`);
+    }
+    if (group.VideoResolution) {
+      conditions.push(`${group.VideoResolution} (${group.VideoWidth}x${group.VideoHeight})`);
+    }
+    if (group.VideoCodecs) {
+      conditions.push(group.VideoCodecs);
+    }
+    return conditions.length ? conditions.join(' | ') : '-';
+  }, [group]);
+  const audioDetails = useMemo(() => {
+    const conditions: string[] = [];
+    if (group.AudioCodecs) {
+      conditions.push(group.AudioCodecs.toUpperCase());
+    }
+    if (group.AudioLanguages) {
+      if (group.AudioStreamCount !== undefined) {
+        conditions.push(`Multi Audio (${group.AudioLanguages.join(', ')})`);
+      } else {
+        conditions.push(group.AudioLanguages.join(', '));
+      }
+    }
+    return conditions.length ? conditions.join(' | ') : '-';
+  }, [group]);
+  const subtitleDetails = useMemo(() => {
+    const conditions: string[] = [];
+    if (group.SubtitleCodecs) {
+      conditions.push(group.SubtitleCodecs.toUpperCase());
+    }
+    if (group.SubtitleLanguages) {
+      if (group.SubtitleStreamCount !== undefined) {
+        conditions.push(`Multi Audio (${group.SubtitleLanguages.join(', ')})`);
+      } else {
+        conditions.push(group.SubtitleLanguages.join(', '));
+      }
+    }
+    return conditions.length ? conditions.join(' | ') : '-';
+  }, [group]);
+  const locationDetails = group.FileLocation ?? '-';
+
+  return (
+    <div className="flex flex-col gap-y-6 rounded border border-panel-border bg-panel-background-transparent p-6">
+      <div className="flex text-xl font-semibold">
+        <Header ranges={group.RangeByType} />
+      </div>
+      <div className="flex gap-x-[4.5rem]">
+        <div className="flex flex-col gap-y-2">
+          <Row label="Group" value={groupDetails} />
+          <Row label="Video" value={videoDetails} />
+          <Row label="Location" value={locationDetails} />
+        </div>
+        <div className="flex flex-col gap-y-2">
+          <Row label="Total" value={sizes} />
+          <Row label="Audio" value={audioDetails} />
+          <Row label="Subtitles" value={subtitleDetails} />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+type Props = {
+  groups?: WebuiSeriesFileSummaryGroupType[];
+};
+const FilesSummaryGroups = ({ groups = [] }: Props) => (
```

Comment:
Needs react.memo, also react uses array index as key by default so this is not doing anything, is there no id we can use?

Reply:
The grouping is dynamic so there's nothing unique in them to add an ID as